openwrt-openwisp-monitoring/files/monitoring.agent

```diff
@@ -219,17 +182,83 @@ send_data() {
 	done
 }
 
-[ -z "$MODE" ] && echoerr "missing required --mode option"
-
-if [ "$MODE" = "collect" ]; then
-	MONITORED_INTERFACES=${MONITORED_INTERFACES:-*}
-	# remove double quotes from interfaces
-	MONITORED_INTERFACES=$(echo "$MONITORED_INTERFACES" | tr -d '"')
-	save_data
-elif [ "$MODE" = "send" ]; then
-	VERIFY_SSL=${VERIFY_SSL:-0}
-	RESPONSE_FILE="$TMP_DIR"/response.txt
-	set_url_and_curl && send_data
-else
-	echoerr "The supplied mode is invalid. Only send and collect are allowed"
-fi
+main() {
+	# parse options
+	while [ -n "$1" ]; do
+		case "$1" in
+			--version | -v)
+				show_version
+				;;
+			--help | -h)
+				show_help
+				;;
+			--url)
+				export BASE_URL="$2"
+				shift
+				;;
+			--uuid)
+				export UUID="$2"
+				shift
+				;;
+			--key)
+				export KEY="$2"
+				shift
+				;;
+			--verify_ssl)
+				export VERIFY_SSL="$2"
+				shift
+				;;
+			--interval)
+				export INTERVAL="$2"
+				shift
+				;;
+			--monitored_interfaces)
+				export MONITORED_INTERFACES="$2"
+				shift
+				;;
+			--verbose_mode)
+				export VERBOSE_MODE="$2"
+				shift
+				;;
+			--required_memory)
+				export REQUIRED_PERCENT="$2"
+				shift
+				;;
+			--mode)
+				export MODE="$2"
+				shift
+				;;
+			--max_retries)
+				export MAX_RETRY="$2"
+				shift
+				;;
+			-*)
+				echo "Invalid option: $1"
+				exit 1
+				;;
+			*) break ;;
+		esac
+		shift
+	done
+
+	INTERVAL=${INTERVAL:-300}
+	VERBOSE_MODE=${VERBOSE_MODE:-0}
+	TMP_DIR="/tmp/openwisp/monitoring"
```

Comment:
Should we not create this dir with `mkdir -p` ?

Reply:
I am doing in *save_data* function
https://github.com/openwisp/openwrt-openwisp-monitoring/blob/7be1724500399ecc0393e4e3fb9ba3fce879c4ec/openwrt-openwisp-monitoring/files/monitoring.agent#L128-L129